messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingExtension.java

```diff
@@ -169,7 +177,7 @@ public static ResourceDescriptionResolver getResourceDescriptionResolver(final b
 
     public void initialize(ExtensionContext context) {
         final SubsystemRegistration subsystemRegistration = context.registerSubsystem(SUBSYSTEM_NAME, CURRENT_MODEL_VERSION);
-        subsystemRegistration.registerXMLElementWriter(new MessagingSubsystemParser_2_0());
+        subsystemRegistration.registerXMLElementWriter(MessagingSubsystemParser_3_0::new);
```

Comment:
Please use new MessagingSubsystemParser_3_0().

The method reference doesn't prevent loading of the class, but it does add more metaspace overhead than a constructed instance adds to the heap.

Reply:
I've no issue with changing that back to instantiating the object.

However, we should update the `registerXMLElementWriter` javadoc:  its `registerXMLElementWriter(XMLElementWriter<SubsystemMarshallingContext> writer)` is `@Deprecated` and recommends to use the variant where we pass a `Supplier` instead.

Reply:
fixed (by amending the WFLY-9407 commit)

Reply:
@jmesnil Good point. https://github.com/wildfly/wildfly-core/pull/2963

Reply:
Well I do not agree with this scenario here.
Writers are rarely used as they are only invoked when there is configuration change, which in production systems happens rarely. This why it makes sense to have them created only on demand instead of up-front.

Parsers on other hand, for boot performance it would somewhat make sense to have "current" parser eagerly created.
But if you do that, parser is not GCable.

Maybe it would make sense to do have another version of registerParser(Supplier, boolean) method which would on register invoke get(), and create the parser before boot process would kick in.
This way we would have eager creation on extension init() as well as having them GC-able.